### ISSUE TITLE
Disable IGMP, there is not igmp upstream interface or igmp downstream interface

### ISSUE THREAD

**Issue by roelvandermade** [fabianishere/udm-iptv#318]
### What is your ISP? (if relevant)

KPN

### Diagnostic Information

```
root@UniFiUXG-Lite:~# udm-iptv-diag
Please share the following output with the developers:
=== Configuration ===
WAN Interface: eth1
WAN VLAN: 4 (dev iptv)
WAN DHCP: true (options "-O staticroutes -V IPTV_RG")
WAN Ranges: 213.75.0.0/16  217.166.0.0/16
LAN Interfaces: eth0.100
IGMP Proxy quickleave disabled: true
IGMP Proxy debug: false
=== IP Link and Route ===
22: iptv@eth1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UNKNOWN group default qlen 1000
    inet 10.173.101.249/20 brd 10.173.111.255 scope global iptv
       valid_lft forever preferred_lft forever
10.173.96.0/20 proto kernel scope link src 10.173.101.249
213.75.112.0/21 via 10.173.96.1 metric 222
=== Service Logs ===
May 01 19:47:35 UniFiUXG-Lite udm-iptvd[2988350]: init_interface: Cannot assign requested address
May 01 19:47:35 UniFiUXG-Lite udm-iptvd[2988350]: ERROR[init_interface@172]: SIOCGIFADDR filed eth0.100
May 01 19:47:35 UniFiUXG-Lite udm-iptvd[2988350]: ERROR[init_interface@172]: SIOCGIFADDR filed eth0
May 01 19:47:35 UniFiUXG-Lite udm-iptvd[2988350]: ERROR[init_interface@232]: Disable IGMP, there is not igmp upstream interface or igmp downstream interface
May 01 19:47:35 UniFiUXG-Lite udm-iptvd[2988350]: FATAL[init_interface@251]: exiting.....
May 01 19:47:35 UniFiUXG-Lite udm-iptvd[2988350]: ERROR[main@619]: exiting.........
May 01 19:47:35 UniFiUXG-Lite IMPROXY[2988350]: ERROR[init_interface@232]: Disable IGMP, there is not igmp upstream interface or igmp downstream interface
May 01 19:47:35 UniFiUXG-Lite systemd[1]: udm-iptv.service: Main process exited, code=exited, status=1/FAILURE
May 01 19:47:35 UniFiUXG-Lite IMPROXY[2988350]: FATAL[init_interface@251]: exiting.....
May 01 19:47:35 UniFiUXG-Lite IMPROXY[2988350]: ERROR[main@619]: exiting.........
```

### Describe the Bug

Am not sure where the exact issue originates from, but since upgrading the Ubiquiti controller software from version 8.1.113 to 8.1.127 (latest stable) my UXG-Lite configuration stopped working with interface errors. Reverting back to version 8.1.113 solves the issue, wondering where this issue might relate to.

### Expected Behavior

Consistent IPTV functionality without erros and stuttering iptv.

**Comment by roelvandermade** [comment]
Active interfaces on the UXG-Lite:

```
root@UniFiUXG-Lite:~# ip a ls
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1
    link/loopback f6:e2:c6:e8:c2:3c brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host
       valid_lft forever preferred_lft forever
2: dummy0: <BROADCAST,NOARP> mtu 1500 qdisc noop state DOWN group default qlen 1000
    link/ether 1e:30:55:b4:66:79 brd ff:ff:ff:ff:ff:ff
3: gre0@NONE: <NOARP> mtu 1476 qdisc noop state DOWN group default qlen 1
    link/gre 0.0.0.0 brd 0.0.0.0
4: gretap0@NONE: <BROADCAST,MULTICAST> mtu 1462 qdisc noop state DOWN group default qlen 1000
    link/ether 00:00:00:00:00:00 brd ff:ff:ff:ff:ff:ff
5: ip_vti0@NONE: <NOARP> mtu 1332 qdisc noop state DOWN group default qlen 1
    link/ipip 0.0.0.0 brd 0.0.0.0
6: ip6_vti0@NONE: <NOARP> mtu 1500 qdisc noop state DOWN group default qlen 1
    link/tunnel6 :: brd ::
7: sit0@NONE: <NOARP> mtu 1480 qdisc noop state DOWN group default qlen 1
    link/sit 0.0.0.0 brd 0.0.0.0
8: ip6tnl0@NONE: <NOARP> mtu 1452 qdisc noop state DOWN group default qlen 1
    link/tunnel6 :: brd ::
9: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq master br0 state UP group default qlen 1000
    link/ether f4:e2:c6:e8:c2:3c brd ff:ff:ff:ff:ff:ff
    inet6 fe80::f6e2:c6ff:fee8:c23c/64 scope link
       valid_lft forever preferred_lft forever
10: eth1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP group default qlen 1000
    link/ether f4:e2:c6:e8:c2:3d brd ff:ff:ff:ff:ff:ff
    inet6 fe80::f6e2:c6ff:fee8:c23d/64 scope link
       valid_lft forever preferred_lft forever
11: ifb0: <BROADCAST,NOARP> mtu 1500 qdisc noop state DOWN group default qlen 32
    link/ether 72:46:eb:41:ec:1f brd ff:ff:ff:ff:ff:ff
12: ifb1: <BROADCAST,NOARP> mtu 1500 qdisc noop state DOWN group default qlen 32
    link/ether f6:51:96:6e:36:0d brd ff:ff:ff:ff:ff:ff
13: br0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether f4:e2:c6:e8:c2:3c brd ff:ff:ff:ff:ff:ff
    inet 192.168.1.1/24 scope global br0
       valid_lft forever preferred_lft forever
    inet6 2a02:a444:54c1::1/64 scope global dynamic
       valid_lft 228213sec preferred_lft 141813sec
    inet6 fe80::f6e2:c6ff:fee8:c23c/64 scope link
       valid_lft forever preferred_lft forever
14: eth0.100@eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master br100 state UP group default qlen 1000
    link/ether f4:e2:c6:e8:c2:3c brd ff:ff:ff:ff:ff:ff
    inet6 fe80::f6e2:c6ff:fee8:c23c/64 scope link
       valid_lft forever preferred_lft forever
15: eth0.110@eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether f4:e2:c6:e8:c2:3c brd ff:ff:ff:ff:ff:ff
    inet 192.168.110.1/24 scope global eth0.110
       valid_lft forever preferred_lft forever
    inet6 fe80::f6e2:c6ff:fee8:c23c/64 scope link
       valid_lft forever preferred_lft forever
16: eth0.150@eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether f4:e2:c6:e8:c2:3c brd ff:ff:ff:ff:ff:ff
    inet 192.168.150.1/24 scope global eth0.150
       valid_lft forever preferred_lft forever
    inet6 fe80::f6e2:c6ff:fee8:c23c/64 scope link
       valid_lft forever preferred_lft forever
17: eth0.200@eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether f4:e2:c6:e8:c2:3c brd ff:ff:ff:ff:ff:ff
    inet 192.168.200.1/24 scope global eth0.200
       valid_lft forever preferred_lft forever
    inet6 fe80::f6e2:c6ff:fee8:c23c/64 scope link
       valid_lft forever preferred_lft forever
18: eth0.220@eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether f4:e2:c6:e8:c2:3c brd ff:ff:ff:ff:ff:ff
    inet 192.168.220.1/28 scope global eth0.220
       valid_lft forever preferred_lft forever
    inet6 fe80::f6e2:c6ff:fee8:c23c/64 scope link
       valid_lft forever preferred_lft forever
19: eth0.500@eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether f4:e2:c6:e8:c2:3c brd ff:ff:ff:ff:ff:ff
    inet 10.100.1.1/24 scope global eth0.500
       valid_lft forever preferred_lft forever
    inet6 2a02:a444:54c1:1::1/64 scope global dynamic
       valid_lft 228213sec preferred_lft 141813sec
    inet6 fe80::f6e2:c6ff:fee8:c23c/64 scope link
       valid_lft forever preferred_lft forever
20: eth1.6@eth1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether f4:e2:c6:e8:c2:3d brd ff:ff:ff:ff:ff:ff
    inet6 fe80::f6e2:c6ff:fee8:c23d/64 scope link
       valid_lft forever preferred_lft forever
21: ppp0: <POINTOPOINT,MULTICAST,NOARP,UP,LOWER_UP> mtu 1492 qdisc fq_codel state UNKNOWN group default qlen 3
    link/ppp
    inet 81.205.XXX.XXX peer 195.190.228.10/32 scope global ppp0
       valid_lft forever preferred_lft forever
    inet6 fe80::8421:1443:XXXX:XXXX/128 scope link
       valid_lft forever preferred_lft forever
22: iptv@eth1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether f4:e2:c6:e8:c2:3d brd ff:ff:ff:ff:ff:ff
    inet6 fe80::f6e2:c6ff:fee8:c23d/64 scope link
       valid_lft forever preferred_lft forever
23: br100: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether f4:e2:c6:e8:c2:3c brd ff:ff:ff:ff:ff:ff
    inet 192.168.100.1/28 scope global br100
       valid_lft forever preferred_lft forever
    inet6 fe80::8861:66ff:fe1f:8e29/64 scope link
       valid_lft forever preferred_lft forever
24: wgsrv1: <POINTOPOINT,NOARP,UP,LOWER_UP> mtu 1420 qdisc noqueue state UNKNOWN group default qlen 1
    link/none
    inet 192.168.2.1/24 scope global wgsrv1
       valid_lft forever preferred_lft forever
```

**Comment by roelvandermade** [comment]
Don't know, messing around with several interfaces got me a working IPTV tv now, but diagnose still shows errors:

```
root@UniFiUXG-Lite:~# udm-iptv diagnose
Please share the following output with the developers:
=== Configuration ===
WAN Interface: eth1
WAN VLAN: 4 (dev iptv)
WAN DHCP: true (options "-O staticroutes -V IPTV_RG")
WAN Ranges: 213.75.0.0/16  217.166.0.0/16
LAN Interfaces: eth0.100
IGMP Proxy quickleave disabled: false
IGMP Proxy debug: false
=== IP Link and Route ===
24: iptv@eth1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    inet 10.173.101.135/20 brd 10.173.111.255 scope global iptv
       valid_lft forever preferred_lft forever
10.173.96.0/20 proto kernel scope link src 10.173.101.135
213.75.112.0/21 via 10.173.96.1 metric 224
=== Service Logs ===
May 01 21:10:13 UniFiUXG-Lite udm-iptvd[29042]: udhcpc: sending discover
May 01 21:10:13 UniFiUXG-Lite udm-iptvd[29042]: udhcpc: sending select for 10.173.101.135
May 01 21:10:13 UniFiUXG-Lite udm-iptvd[29042]: udhcpc: lease of 10.173.101.135 obtained, lease time 3600
May 01 21:10:13 UniFiUXG-Lite udm-iptvd[29030]: NATing IPTV network ranges (if necessary)
May 01 21:10:13 UniFiUXG-Lite udm-iptvd[29030]: Creating static routes (if necessary)
May 01 21:10:13 UniFiUXG-Lite udm-iptvd[29030]: Setting up IGMP Proxy
May 01 21:10:13 UniFiUXG-Lite udm-iptvd[29030]: Using improxy...
May 01 21:10:13 UniFiUXG-Lite udm-iptvd[29030]: Starting IGMP Proxy
May 01 21:10:14 UniFiUXG-Lite IMPROXY[29030]: ERROR[init_interface@172]: SIOCGIFADDR filed eth0
May 01 21:10:14 UniFiUXG-Lite udm-iptvd[29030]: init_interface: Cannot assign requested address
root@UniFiUXG-Lite:~#
```

**Comment by mauriceberentsen** [comment]
`Please share the following output with the developers:
=== Configuration ===
WAN Interface: eth8
WAN VLAN: 4 (dev iptv)
WAN DHCP: true (options "-O staticroutes -V IPTV_RG")
WAN Ranges: 213.75.0.0/16  217.166.0.0/16
LAN Interfaces: br40
IGMP Proxy quickleave disabled: false
IGMP Proxy debug: true
=== IP Link and Route ===
=== Service Logs ===
May 04 08:38:15 Dream-Machine-Pro-Max IMPROXY[228503]: ERROR[main@619]: exiting.........
May 04 08:38:15 Dream-Machine-Pro-Max systemd[1]: udm-iptv.service: Failed with result 'exit-code'.
May 04 08:38:20 Dream-Machine-Pro-Max systemd[1]: udm-iptv.service: Scheduled restart job, restart counter is at 3.
May 04 08:38:20 Dream-Machine-Pro-Max systemd[1]: Stopped IPTV support for the UniFi Dream Machine.
May 04 08:38:20 Dream-Machine-Pro-Max systemd[1]: Started IPTV support for the UniFi Dream Machine.
May 04 08:38:20 Dream-Machine-Pro-Max udm-iptvd[229534]: Device iptv already exists.. Deleting device
May 04 08:38:20 Dream-Machine-Pro-Max udm-iptvd[229534]: Obtaining IP address for VLAN interface
May 04 08:38:20 Dream-Machine-Pro-Max udm-iptvd[229548]: udhcpc: started, v1.30.1
May 04 08:38:20 Dream-Machine-Pro-Max udm-iptvd[229548]: udhcpc: sending discover
May 04 08:38:23 Dream-Machine-Pro-Max udm-iptvd[229548]: udhcpc: sending discover
root@Dream-Machine-Pro-Max:~# udm-iptv diagnose
Please share the following output with the developers:
=== Configuration ===
WAN Interface: eth8
WAN VLAN: 4 (dev iptv)
WAN DHCP: true (options "-O staticroutes -V IPTV_RG")
WAN Ranges: 213.75.0.0/16  217.166.0.0/16
LAN Interfaces: br40
IGMP Proxy quickleave disabled: false
IGMP Proxy debug: true
=== IP Link and Route ===
=== Service Logs ===
May 04 08:38:30 Dream-Machine-Pro-Max udm-iptvd[229534]: DEBUG[init_interface@178]: br40's ipv4 address is 192.168.40.1
May 04 08:38:30 Dream-Machine-Pro-Max udm-iptvd[229534]: DEBUG[init_interface@195]: if_index = 34
May 04 08:38:30 Dream-Machine-Pro-Max udm-iptvd[229534]: DEBUG[init_interface@208]: if_mtu = 1500
May 04 08:38:30 Dream-Machine-Pro-Max udm-iptvd[229534]: DEBUG[init_interface@214]: if_falgs = 4163
May 04 08:38:30 Dream-Machine-Pro-Max udm-iptvd[229534]: ERROR[init_interface@172]: SIOCGIFADDR filed iptv
May 04 08:38:30 Dream-Machine-Pro-Max udm-iptvd[229534]: ERROR[init_interface@232]: Disable IGMP, there is not igmp upstream interface or igmp downstream interface
May 04 08:38:30 Dream-Machine-Pro-Max udm-iptvd[229534]: FATAL[init_interface@251]: exiting.....
May 04 08:38:30 Dream-Machine-Pro-Max udm-iptvd[229534]: ERROR[main@619]: exiting.........
May 04 08:38:30 Dream-Machine-Pro-Max systemd[1]: udm-iptv.service: Main process exited, code=exited, status=1/FAILURE
May 04 08:38:30 Dream-Machine-Pro-Max systemd[1]: udm-iptv.service: Failed with result 'exit-code'.`

I got the same issue, I think. Could you elaborate on your solution?

**Comment by Sebazzz** [comment]
I’m also curious. @roelvandermade

**Comment by roelvandermade** [comment]
I understand the curiosity because I also found that there's very little in information to find with the issue I had
Like I said above, I was fiddling around  with the settings mentioned in Fabian's documentation and somehow it started working again after reboots and lots of trials..  sorry can't be of any further help here.

These are my current working settings (even though it shows an error in the diag output), I have a separate IPTV vlan100, hence the eth0.100 ipv interface:

```
root@UniFiUXG-Lite:~# udm-iptv diagnose
Please share the following output with the developers:
=== Configuration ===
WAN Interface: eth1
WAN VLAN: 4 (dev iptv)
WAN DHCP: true (options "-O staticroutes -V IPTV_RG")
WAN Ranges: 213.75.0.0/16  217.166.0.0/16
LAN Interfaces: eth0.100
IGMP Proxy quickleave disabled: false
IGMP Proxy debug: false
=== IP Link and Route ===
23: iptv@eth1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    inet 10.173.111.63/20 brd 10.173.111.255 scope global iptv
       valid_lft forever preferred_lft forever
10.173.96.0/20 proto kernel scope link src 10.173.111.63
213.75.112.0/21 via 10.173.96.1 metric 223
=== Service Logs ===
Sep 02 18:13:17 UniFiUXG-Lite udm-iptvd[1619732]: udhcpc: sending discover
Sep 02 18:13:17 UniFiUXG-Lite udm-iptvd[1619732]: udhcpc: sending select for 10.173.111.63
Sep 02 18:13:17 UniFiUXG-Lite udm-iptvd[1619732]: udhcpc: lease of 10.173.111.63 obtained, lease time 3600
Sep 02 18:13:17 UniFiUXG-Lite udm-iptvd[1619716]: NATing IPTV network ranges (if necessary)
Sep 02 18:13:17 UniFiUXG-Lite udm-iptvd[1619716]: Creating static routes (if necessary)
Sep 02 18:13:17 UniFiUXG-Lite udm-iptvd[1619716]: Setting up IGMP Proxy
Sep 02 18:13:17 UniFiUXG-Lite udm-iptvd[1619716]: Using improxy...
Sep 02 18:13:17 UniFiUXG-Lite udm-iptvd[1619716]: Starting IGMP Proxy
Sep 02 18:13:17 UniFiUXG-Lite IMPROXY[1619716]: ERROR[init_interface@172]: SIOCGIFADDR filed eth0
Sep 02 18:13:17 UniFiUXG-Lite udm-iptvd[1619716]: init_interface: Cannot assign requested address
```

**Comment by Sebazzz** [comment]
I kind-of figured it out but I can’t tell why it works, because by all accounts it should not work. I have the Sagemcom DIW7022 Android-TV based KPN TV+ box. It worked fine on wifi, but complained in the diagnostics that multicast didn’t work. When I finally got around to cleaning up the cabling in my TV cabinet, I switched the KPN TV+ box to wired ethernet, but then it actually started failing (frozen TV) when I used multicast. 

Without touching udm-iptv (it was already in the failed state I showed in my comment), I just enabled the UDM native IGMP proxy, and the TV box started working, and the built-in diagnosis shows multicast is working fine. 

Note: because I still have DSL my setup is actually:

   DSL point <—RJ12–> Experiabox <—ethernet-DMZ—> UDM <—wired ethernet —> Unifi 8-port <— ethernet —> KPN TV+ box

So in summary: I installed the udm-iptvd but it crashes/exits, and later enabled the UDM native IGMP proxy which works fine.

**Comment by cbundy** [comment]
Same situation here. Mine is also failing, but I have TV working with the native forwarding.

Has anyone removed this application to see if it still works without it at all? I'm tempted but I just got it working and I'd rather leave it for now.

If it works natively then it might be worth updating the documentation to point out that for KPN this is no longer needed, to save people hassle in the future

**Comment by Taykun345** [comment]
ITS seems you dont need udm-iptv anymore (at least for providers that do not use VLAN) on new Unifi OS. With IGMP proxy enabled, it just works.

**Comment by wvdheiden** [comment]
I noticed that the latest sofwtare of teh dreammachine has a button "IGMP proxy" ('bout effing time....)
Just click it, select your IPTV network and bäm: you're good to go.
You can do: udm-iptv uninstall to uninstall Fabian's masterpiece.

Thank you, Fabian! You are the best.

**Comment by rogierlommers** [comment]
> I noticed that the latest sofwtare of teh dreammachine has a button "IGMP proxy" ('bout effing time....) Just click it, select your IPTV network and bäm: you're good to go. You can do: udm-iptv uninstall to uninstall Fabian's masterpiece.
> 
> Thank you, Fabian! You are the best.

Where exactly do you see that button? Can you post a screenshot?

**Comment by dmutsaers** [comment]
> ...at least for providers that do not use VLAN) on new Unifi OS

A quite important detail

**Comment by rogierlommers** [comment]
> > ...at least for providers that do not use VLAN
> 
> A very important detail. The IGMP Proxy has been available quite some time.

Ah ideed... I just updated the OS by the way (version 4.1.22).
Then re-installed the script again (KPN).
Evertything works fine.

**Comment by wvdheiden** [comment]
Yes sure. 

![Image](https://github.com/user-attachments/assets/3fdf2f4d-b389-43d4-b18f-d9941126a399)
![Image](https://github.com/user-attachments/assets/6cac7afe-1c77-49b4-b57c-a9dc8ae1f267)

I never reaized IPTV streamiing was in there longer. I am using a UDM for a long time. I stopped looking and hoping quite a while ago. (It was working)
I noticed that channel changing is much slower with the UDM option on. Not sure if that is KPN or the UDM.

**Comment by wvdheiden** [comment]
Oooohhhhkkkkkaaaayyyy?? Hmmm Yes I do use the new one and I'm not happy. Lots of hangups, bad synchronization with sound/image, slow zap-time. 
Is there no "built-in" possibility to use the UDM thing? (I am not dumb but also not an expert)
(thanks for your input BTW)

**Comment by wvdheiden** [comment]
![Image](https://github.com/user-attachments/assets/10653bf3-3fb6-4cba-bced-5bb332304faa)
And there is this. Are these the correct settings?

**Comment by dmutsaers** [comment]
That looks OK to me (if you disable the native IGMP Proxy and install and configure UDM-IPTV). After every firmware update you need to reinstall UDM-IPTV. (And all of your switches need to support IGMP Snooping)

**Comment by dmutsaers** [comment]
Fast leave can be enabled too.

**Comment by wvdheiden** [comment]
Thank you for your help. Really nice. 
I switched on "Fast Leave." It asks for which VLAN this should be set (IPTV 661 in my case) 
I disabled the IGMP (Grrr) and installed back the good ol' proxy. 
I am aware that it must be reinstalled after each update. But that's OK. I can live with that. I don't have Automotic updates on anyway. So I can plan updates. I sometimes feel like an IT department here at home...
What is "Forward unknown Multicast Traffic"?

**Comment by wvdheiden** [comment]
And then there is the other question: 
[ ] IGMPv2
[ ] IGMPv3

I think I always used v2 so I stick to that one. Or should I change? Since today I know my TV will work without IGMP Proxy so all my previous "trial and error" is no longer vald.

**Comment by fabianishere** [comment]
@roelvandermade This error occurs when the upstream interfaces do not have any IP address associated with them. In the example you shared, both eth0 and eth0.100 do not have any IP addresses. Are you sure that interface is associated with any of your VLANs?

Given that it has been a long while since this issue was opened, I'll close it for now since the thread seems to have diverged a bit. Feel free to re-open if the above doesn't help.
